src/modules/preferences/valueAccordion/useValueAccordion.tsx

```diff
@@ -0,0 +1,54 @@
+import { useEffect, useState } from 'react';
+
+import { ValueAccordionItem } from './valueAccordion.types';
+import { ValueGroup } from '../valueContainer/valueContainer.types';
+
+export const useValueAccordion = (
+  valueGroup: ValueGroup,
+  items: ValueAccordionItem[],
+  setItems: (value: ValueAccordionItem[]) => void,
+  setError: (valueGroup: ValueGroup, error: string) => void,
+) => {
+  const [letterCount, setletterCount] = useState(
+    items.find(item => item.valueGroup === valueGroup && item.key.includes('OTHERS'))?.description?.length || 0,
+  );
+
+  const handleChange = (itemKey: string) => {
+    const lst = [...items];
+
+    const item = lst.find(item => item.key === itemKey);
+    if (!item) return;
+    else {
+      const newVal = item.value === 'ON' ? 'OFF' : 'ON';
+      Object.assign(item, { ...item, value: newVal });
```

Comment:
FMI: i can't understand why you didn't use `map` function on `items`
to change value to `newValue` for your condition

Reply:
line 43 too